### PR TITLE
added qunit testing to project runs with make

### PR DESCRIPTION
**Pull request by coati-00**

### CODE REVIEW

**Comment by thraxil** [comment]
It looks like you checked in the entire `node_modules` subdirectory.

**Comment by nikolas** [comment]
Yeah, we don't need to include all those dependencies. See how DMT's `.gitignore` ignores the `node_modules` directory: https://github.com/ccnmtl/dmt/blob/master/.gitignore

File: media/js/ccnmtljs/bb_mixins.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
+/* Putting Mixins to use for validation in multiple views here */
+
+// A simple object with some methods
+var ValidationMixin = {
```
Comment (nikolas) [inline media/js/ccnmtljs/bb_mixins.js]: These validators are cool - I haven't used mixins in plain JavaScript before (without Ember.js's classes) and it seems like a nice way to organize it.

**Comment by nikolas** [comment]
I haven't run these tests yet, but one suggestion I have so far is to squash the commits that contain all the `node_modules` dependencies so this branch doesn't increase the repository's size with unnecessary objects in its history.

I can help you do this when you get a chance, and there's a write-up of `git rebase -i` here: http://wiki.ccnmtl.columbia.edu/index.php/Git#Re-organizing_your_commits

**Pull request merged**